src/iceberg/catalog/rest/rest_catalog.cc

```diff
@@ -115,29 +117,67 @@ Result<std::vector<Namespace>> RestCatalog::ListNamespaces(const Namespace& ns)
 }
 
 Status RestCatalog::CreateNamespace(
-    [[maybe_unused]] const Namespace& ns,
-    [[maybe_unused]] const std::unordered_map<std::string, std::string>& properties) {
-  return NotImplemented("Not implemented");
+    const Namespace& ns, const std::unordered_map<std::string, std::string>& properties) {
+  ICEBERG_ASSIGN_OR_RAISE(auto endpoint, paths_->Namespaces());
+  CreateNamespaceRequest request{.namespace_ = ns, .properties = properties};
+  ICEBERG_ASSIGN_OR_RAISE(auto json_request, ToJsonString(ToJson(request)));
+  ICEBERG_ASSIGN_OR_RAISE(const auto& response,
+                          client_->Post(endpoint, json_request, /*headers=*/{},
+                                        *NamespaceErrorHandler::Instance()));
+  ICEBERG_ASSIGN_OR_RAISE(auto json, FromJsonString(response.body()));
+  ICEBERG_ASSIGN_OR_RAISE(auto create_response, CreateNamespaceResponseFromJson(json));
+  return {};
 }
 
 Result<std::unordered_map<std::string, std::string>> RestCatalog::GetNamespaceProperties(
-    [[maybe_unused]] const Namespace& ns) const {
-  return NotImplemented("Not implemented");
-}
-
-Status RestCatalog::DropNamespace([[maybe_unused]] const Namespace& ns) {
-  return NotImplemented("Not implemented");
+    const Namespace& ns) const {
+  ICEBERG_ASSIGN_OR_RAISE(auto endpoint, paths_->Namespace_(ns));
+  ICEBERG_ASSIGN_OR_RAISE(const auto& response,
+                          client_->Get(endpoint, /*params=*/{}, /*headers=*/{},
+                                       *NamespaceErrorHandler::Instance()));
+  ICEBERG_ASSIGN_OR_RAISE(auto json, FromJsonString(response.body()));
+  ICEBERG_ASSIGN_OR_RAISE(auto get_response, GetNamespaceResponseFromJson(json));
+  return get_response.properties;
 }
 
-Result<bool> RestCatalog::NamespaceExists([[maybe_unused]] const Namespace& ns) const {
-  return NotImplemented("Not implemented");
+Status RestCatalog::DropNamespace(const Namespace& ns) {
+  ICEBERG_ASSIGN_OR_RAISE(auto endpoint, paths_->Namespace_(ns));
+  ICEBERG_ASSIGN_OR_RAISE(
+      const auto& response,
+      client_->Delete(endpoint, /*headers=*/{}, *DropNamespaceErrorHandler::Instance()));
+  return {};
+}
+
+Result<bool> RestCatalog::NamespaceExists(const Namespace& ns) const {
+  ICEBERG_ASSIGN_OR_RAISE(auto endpoint, paths_->Namespace_(ns));
+  auto response_or_error =
```

Comment:
This impl is not consistent with Java where backward compatibility is considered to use GetNamespaceProperties as a work around.